translator/context/context.go

```diff
@@ -128,7 +133,23 @@ func (ctx *Context) SetMode(mode string) {
 		ctx.mode = config.ModeWithIRSA
 		ctx.shortMode = config.ShortModeWithIRSA
 	default:
-		log.Panicf("Invalid mode %s. Valid mode values are %s, %s, %s and %s.", mode, config.ModeEC2, config.ModeOnPrem, config.ModeOnPremise, config.ModeWithIRSA)
+		log.Panicf("Invalid mode %s. Valid mode values are %s, %s, %s, and %s.", mode, config.ModeEC2, config.ModeOnPrem, config.ModeOnPremise, config.ModeWithIRSA)
+	}
+}
+
+func (ctx *Context) SetKubernetesMode(mode string) {
+	switch mode {
+	case config.ModeEKS:
+		ctx.kubernetesMode = config.ModeEKS
+		ctx.shortMode = config.ShortModeEKS
```

Comment:
So in this case, the shortMode will be `EKS` but the mode would be `EC2` - would that lead to any issues?

Reply:
Yeah, I spoke to Jeffrey and he mentioned we only use shortMode for our AgentHealth metrics

Reply:
Got it. Not a blocker but perhaps we should just rename this to userAgentMode or something.
mode & shortMode sets an implicit expectation that they will be in sync and we might add future usages of this shortMode without realizing it gets overwritten here.

Reply:
Alternatively, can we set both shortMode & kubernetesMode on the header so we dont do this override here?
Not sure if thats going too far to prevent this.

Reply:
That would require changes to the frontend.